static/app/components/forms/fields/segmentedRadioField.tsx

```diff
@@ -0,0 +1,179 @@
+import isPropValid from '@emotion/is-prop-valid';
+import styled from '@emotion/styled';
+
+import type {RadioGroupProps} from 'sentry/components/forms/controls/radioGroup';
+import type {InputFieldProps, OnEvent} from 'sentry/components/forms/fields/inputField';
+import FormField from 'sentry/components/forms/formField';
+import InteractionStateLayer from 'sentry/components/interactionStateLayer';
+import Radio from 'sentry/components/radio';
+import {Tooltip} from 'sentry/components/tooltip';
+import {space} from 'sentry/styles/space';
+
+export interface SegmentedRadioFieldProps extends Omit<InputFieldProps, 'type'> {
+  choices?: RadioGroupProps<any>['choices'];
+}
+
+function handleChange(
+  id: string,
+  onChange: OnEvent,
+  onBlur: OnEvent,
+  e: React.FormEvent<HTMLInputElement>
+) {
+  onChange(id, e);
+  onBlur(id, e);
+}
+
+function SegmentedRadioField(props: SegmentedRadioFieldProps) {
+  return (
+    <FormField {...props}>
+      {({id, onChange, onBlur, value, disabled, ...fieldProps}) => (
+        <ControlGroup
+          id={id}
+          name={props.name}
+          choices={fieldProps.choices}
+          disabled={disabled}
+          label={fieldProps.label}
+          value={value === '' ? null : value}
+          onChange={(v, e) => handleChange(v, onChange, onBlur, e)}
+        />
+      )}
+    </FormField>
+  );
+}
+
+function ControlGroup<C extends string>({
+  name: groupName,
+  value,
+  disabled: groupDisabled,
+  disabledChoices = [],
+  choices = [],
+  label,
+  onChange,
+  tooltipPosition,
+  ...props
+}: RadioGroupProps<C>) {
+  return (
+    <Container {...props} role="radiogroup" aria-label={label}>
+      {choices.map(([id, name, description], index) => {
+        const disabledChoice = disabledChoices.find(([choiceId]) => choiceId === id);
+        const disabledChoiceReason = disabledChoice?.[1];
```

Comment:
🤔 i'm guessing we're limited to this being a tuple by the `RadioOption` type definition? (In general, i think an object here might be a little easier to grok / read, but nbd if it is limited by that super low level type)